public/app/plugins/datasource/alertmanager/plugin.json

```diff
@@ -4,42 +4,77 @@
   "id": "alertmanager",
   "metrics": false,
   "routes": [
+    {
+      "method": "GET",
+      "path": "alertmanager/api/v2/silences",
+      "reqRole": "Viewer",
+      "reqAction": "alert.instances.external:read"
+    },
     {
       "method": "POST",
       "path": "alertmanager/api/v2/silences",
-      "reqRole": "Editor"
+      "reqRole": "Editor",
+      "reqAction": "alert.instances.external:write"
+    },
+    {
+      "method": "GET",
+      "path": "alertmanager/api/v2/silence",
+      "reqRole": "Viewer",
+      "reqAction": "alert.instances.external:read"
     },
     {
       "method": "DELETE",
       "path": "alertmanager/api/v2/silence",
-      "reqRole": "Editor"
+      "reqRole": "Editor",
```

Comment:
let's add slash at the end of `alertmanager/api/v2/silence` to reduce the scope when `HasPrefix` is positive.